### PR TITLE
misc updates on dsproof packet

### PR DESCRIPTION
**Pull request by markblundeberg**
- tx_dig -> tx_pre
- include lengths

### CODE REVIEW

**Comment by imaginaryusername** [comment]
Since the document has been updated for multisig, I'll address the renaming a bit later.

File: dsproof/dsproof.md [hunk 1]
```diff
@@ -38,12 +38,14 @@ If the two transaction share more than one outpoint, then a corresponding number
 | Field Size | Description | Data Type  | Comments |
 | -----------|:-----------:| ----------:|---------:|
 | 1 | version | uint8 | version byte |
-| ? | tx_dig1 | tx_dig | Transaction digest of the one transaction before final double SHA256 hash, corresponding to the doublespent outpoint|
-| ? | tx_dig2 | tx_dig | Transaction digest of a different transaction before final double SHA256 hash, corresponding to the doublespent outpoint|
-| ? | sig1 | char | signature of the double-SHA256 of tx_dig1, extracted from tx1 |
-| ? | sig2 | char | signature of the double-SHA256 of tx_dig2, extracted from tx2 |
+| 184 | tx_pre1 | tx_sig_preimage | Signed data from tx1's CHECKSIG execution (includes outpoint). |
+| 184 | tx_pre2 | tx_sig_preimage | Signed data from tx1's CHECKSIG execution (includes outpoint). |
+| 1 | len1 | uint8 | length of sig1 |
```
Comment (zander) [inline dsproof/dsproof.md:43]: Using 'uint8' for length isn't what is used in most of the p2p protocol. I suggest using var-int (confusingly in code it's called CompactSize).

The on-the-wire effect is still one byte in practise.

Comment (markblundeberg) [reply dsproof/dsproof.md:43]: Good point, it's probably best to use var_ints for structured data like this, just for consistency (even though I think they're icky :-D )

**Comment by imaginaryusername** [comment]
Cleaned up some conflicts, merging / dealing with the p2sh part separately.

**Pull request merged**